### PR TITLE
Automatic discovery of Connectors in Job Worker Run-Time based

### PR DESCRIPTION
**Pull request by nikku**
## Description

Adds auto-discovery of connectors to the job worker run-time:

* instantiates all `OutboundConnectorFunction` SPI providers
* picks up our new `@OutboundConnector` annotation (https://github.com/camunda/connector-sdk/issues/12#issuecomment-1254830619).
* manual configuration mode remains un-touched (we'll either use automatic _OR_ manual mode)
* changed prefix for connector related ENV variables from `ZEEBE_` to `CONNECTOR_`

## Related issues

Closes https://github.com/camunda/connector-sdk/issues/12

### CODE REVIEW

**Comment by nikku** [comment]
From what it looks like I'd need to bootstrap a full fledged integration test to verify the behavior (cannot mock System#getEnv): https://github.com/camunda/connector-sdk/compare/connector-auto-discovery-test?expand=1

File: runtime-job-worker/src/main/java/io/camunda/connector/runtime/jobworker/Main.java [hunk 1]
```diff
@@ -66,23 +64,21 @@ public static void main(String[] args) {
...
+                  registration -> {
                     LOGGER.info(
-                        "Registering connector function {} as {} on job type {} with variables {}",
-                        connector.getFunction(),
-                        connector.getName(),
-                        connector.getType(),
-                        connector.getVariables());
-
-                    var connectorFunction = loadConnectorFunction(connector.getFunction());
+                        "Registering registration function {} as {} on job type {} with variables {}",
```
Comment (tmetzke) [inline runtime-job-worker/src/main/java/io/camunda/connector/runtime/jobworker/Main.java:69]:
```suggestion
                        "Registering Connector function {} as {} on job type {} with variables {}",
```

File: runtime-job-worker/src/main/java/io/camunda/connector/runtime/jobworker/impl/outbound/OutboundConnectorRegistration.java [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+        | InstantiationException
+        | IllegalAccessException
+        | ClassCastException
+        | NoSuchMethodException e) {
+      throw loadFailed("Failed to load " + clsName, e);
+    }
+  }
+
+  private static RuntimeException loadFailed(String s, Exception e) {
+    throw new IllegalStateException(s, e);
```
Comment (tmetzke) [inline runtime-job-worker/src/main/java/io/camunda/connector/runtime/jobworker/impl/outbound/OutboundConnectorRegistration.java:184]:
```suggestion
    return new IllegalStateException(s, e);
```

**Review by tmetzke — APPROVED** [review]
This looks good to me so far 👍 
Regarding the ENV vars, we could use [System Stubs](https://www.baeldung.com/java-system-stubs), for example. I won't be able to look into this until later tonight. @nikku, if you have time and want to go ahead, feel free. Otherwise, I'll look into it this evening (CEST).

**Comment by nikku** [comment]
Pushed test coverage on top (https://github.com/camunda/connector-sdk/pull/111/commits/e6506b4d1bde39713ceda29ad7c52cd2036d04f0). Not sure how that magic utility does it, but it seems to get the trick done.

File: runtime-job-worker/pom.xml [hunk 3]
```diff
@@ -51,6 +55,12 @@
       <scope>test</scope>
     </dependency>
 
+    <dependency>
+      <groupId>uk.org.webcompere</groupId>
+      <artifactId>system-stubs-core</artifactId>
+      <version>${system-stubs.version}</version>
```
Comment (tmetzke) [inline runtime-job-worker/pom.xml:61]: Let's add test scope here.
```suggestion
      <version>${system-stubs.version}</version>
      <scope>test</test>
```

Comment (nikku) [reply runtime-job-worker/pom.xml:61]: Cf. https://github.com/camunda/connector-sdk/pull/111/commits/30dbc88ae1a50b49783d9f444bde86ce7d0fb5ea.

File: runtime-job-worker/src/main/java/io/camunda/connector/runtime/jobworker/impl/outbound/OutboundConnectorRegistration.java (hunk 2 above)
Comment (nikku) [reply runtime-job-worker/src/main/java/io/camunda/connector/runtime/jobworker/impl/outbound/OutboundConnectorRegistration.java]: Cf. https://github.com/camunda/connector-sdk/pull/111/commits/30dbc88ae1a50b49783d9f444bde86ce7d0fb5ea.

**Comment by nikku** [comment]
Squashed / cleaned up via https://github.com/camunda/connector-sdk/pull/111/commits/d03baa693e3b869e75438fcea1fae3539a386532.

**Pull request merged**